packages/core/test/endowmentsToolkit.spec.js

```
@@ -144,6 +145,75 @@ test('instrumentDynamicValueAtPath puts a getter at path', (t) => {
   t.is(target.a.b.c, 2)
 })
 
+test('copyWrappedGlobals - writable fields get updated in compartments and in root but not globally', (t) => {
+  const knownWritable = new Set(['a'])
+  const { getEndowmentsForConfig, copyWrappedGlobals, endowAll } = prepareTest({
+    knownWritable,
+  })
+
+  const originalSource = { a: 1, b: 2 }
+
+  const config = {
+    globals: {
+      a: 'write',
+    },
+  }
+  // A way to separate the root compartment endowment from the store for globals
+  // in lavamoat. Note that endowAll uses the same optional references for unwrapping
+
+  // Storage for globals we cache in lavamoat
+  const goldenMasterRecordGlobal = copyWrappedGlobals(originalSource, {})
+  // root compartment global
+  const rootGlobal = endowAll(goldenMasterRecordGlobal)
+  // non-root compartments
+  const endowments = getEndowmentsForConfig(goldenMasterRecordGlobal, config)
+  const endowments2 = getEndowmentsForConfig(goldenMasterRecordGlobal, config)
+  endowments.a = 42
```

Would it be valuable to have a failure case for this?  e.g., when

```js
const config = {
  globals: {
    a: true
  }
}
```

---

I added the test and noticed a minor compatibility issue.
If a field is writable elsewhere, an attempt to write to it in a compartment that is in strict mode and had it endowed as readonly is going to cause a `Cannot set property a of #<Object> which has only a getter` error.

I'm wondering whether it's a good thing to have an error (maybe a setter that throws a LavaMoat specific one) or should we let it overwrite the local copy by creating a setter that replaces itself (and the getter) with a value when invoked.

That would be a more compatible behavior - effectively identical to `a` not being writable elsewhere - but a silent failure when you have code attempting to write to a global that's already writable elsewhere and it might help in debugging.